### ISSUE TITLE
Citation auto-completion doesn't work properly

### ISSUE THREAD

**Issue by thmmsn** [SublimeText/LaTeXTools#1338]
Auto-pairing curly brackets restrict the citation pop-up from beeing triggered.

If I type \cite{ the closing brackets shows up ( \cite{} automatically) and not the citation box. 

If I first type \citea and then move the cursor between e and a, \cite|a ,and then add the curly bracket \cite{a the citation popup shows up. The reference is added and I have to remove the letter 'a'.

**Comment by r-stein** [comment]
Have you added the auto pairing to your user keybindings?

**Comment by thmmsn** [comment]
No, just a default installation of Latextools and the newest version of sublime. Everything was fine until the latest sublime version. Ive also reinstalled latextools.

**Comment by r-stein** [comment]
That weird. You may open the ST console ``ctrl+` `` and run `sublime.log_commands(True)` and `sublime.log_input(True)` and then try to press the button again.

**Comment by thmmsn** [comment]
First case: The citation popup does not show up, but after typing `test` the console return that a Bib file is found.

`
chr evt: \ (0x5c)
chr evt: c (0x63)
chr evt: i (0x69)
chr evt: t (0x74)
chr evt: e (0x65)
key evt: shift+alt+8
chr evt: { (0x7b)
command: insert_snippet {"contents": "{$0}"}
chr evt: t (0x74)
chr evt: e (0x65)
chr evt: s (0x73)
chr evt: t (0x74)
TEX root: 'main.tex'
Bib files found: 
('lib.bib',)
`


Second case: The citation popup shows up.

`
chr evt: \ (0x5c)
chr evt: c (0x63)
chr evt: i (0x69)
chr evt: t (0x74)
chr evt: e (0x65)
chr evt: l (0x6c)
key evt: left
key evt: left
command: move {"by": "characters", "forward": false}
key evt: shift+alt+8
chr evt: { (0x7b)
command: latex_fill_all {"completion_type": "cite", "insert_char": "{"}
TEX root: 'main.tex'
Bib files found: 
('lib.bib',)
`

**Comment by deathaxe** [comment]
The expected sequence should look like

```
key evt: ctrl+alt+[
chr evt: \ (0x5c)
chr evt: c (0x63)
chr evt: i (0x69)
chr evt: t (0x74)
chr evt: e (0x65)
key evt: ctrl+alt+7
chr evt: { (0x7b)
command: latextools_fill_all {"completion_type": "cite", "insert_char": "{"}
```

`insert_snippet` being triggered indicates required key binding to be overridden or context not being detected, correctly.

I can't reproduce this issue with LaTeXTools v4. Hence assuming fixed or not caused by LaTeXTools
